src/core/InstrumentFunctions.cpp

```diff
@@ -408,7 +408,37 @@ void InstrumentFunctionArpeggio::processNote( NotePlayHandle * _n )
 			continue;
 		}
 
-		const int dir = m_arpDirectionModel.value();
+		// Skip notes randomly
+		if( m_arpSkipModel.value() )
+		{
+
+			if( 101 * ( (float) rand() / (float) RAND_MAX ) < m_arpSkipModel.value() )
```

Comment:
Yeah. I was trying to find a more interesting response from the knobs. I've reverted to the same algorithm for both functions. Keep it simple, etc...